src/services/summary/hooks/trigger-calculation.js

```diff
@@ -0,0 +1,88 @@
+const _ = require('lodash');
+const moment = require('moment');
+const ActivityLog = require('../../activity-log/activity-log-model');
+const actionConstants = require('../../../constants/actions');
+const Summary = require('../../summary/summary-model');
+
+// triggerCalculation is a before hook function to trigger a summary
+// calculation on specified dates and organization.
+// If no date specifies, it will calculate today's summary.
+const triggerCalculation = () => (hook) => {
+  const startDate = hook.params.query.start_date;
+  const endDate = hook.params.query.end_date;
+  const organizationName = hook.params.query.organization;
+  const trigger = hook.params.query.trigger;
+  if (!trigger) {
+    return Promise.resolve(hook);
+  }
+  let timestamp = {};
+  // Change start_date and end_date to mongoose query format.
+  if (startDate) {
+    timestamp.$gte = startDate;
+  }
+  if (endDate) {
+    timestamp.$lte = endDate;
+  }
+  if (_.isEmpty(timestamp)) {
+    timestamp = moment().format('YYYY-MM-DD');
+  }
+  // Start a calculation
+  const query = {
+    organization: organizationName,
+    timestamp,
+  };
+  // Remap state to summary's field name.
+  const remapStateToFieldName = {};
+  remapStateToFieldName[actionConstants.ASSIGN] = 'assigned';
+  remapStateToFieldName[actionConstants.PROCESS] = 'processing';
+  remapStateToFieldName[actionConstants.RESOLVE] = 'resolved';
+  return ActivityLog.find(query)
+    .then(logs => {
+      // Loop through activity log and summarize each day into table.
+      const summaryTable = {};
+      for (let i = 0, len = logs.length; i < len; ++i) {
```

Comment:
I just don't see the benefit of creating a new variable in this case. (Normally, I prefer i++ also but it doesn't matter)